101-azure-database-migration-service/azuredeploy.json

```diff
@@ -137,81 +112,193 @@
-            "sku":
-            {
+            "sku": {
                 "name": "GeneralPurpose_4vCores",
                 "tier": "General Purpose",
                 "size": "4 vCores"
             },
             "name": "[variables('DMSServiceName')]",
             "apiVersion": "2017-11-15-privatepreview",
             "location": "[parameters('location')]",
-            "properties":
-            {
+            "properties": {
                 "virtualSubnetId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVNet'), variables('defaultSubnetName'))]"
             },
+            "resources": [
+                {
+                    "type": "projects",
+                    "name": "SqlToSqlDbMigrationProject",
+                    "apiVersion": "2017-11-15-privatepreview",
+                    "location": "[parameters('location')]",
+                    "properties": {
+                        "sourcePlatform": "SQL",
+                        "targetPlatform": "SQLDB"
+                    },
+                    "dependsOn": [
+                        "[resourceId('Microsoft.DataMigration/services', variables('DMSServiceName'))]"
+                    ]
+                }
+            ],
             "dependsOn": [
                 "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVNet'), variables('defaultSubnetName'))]"
             ]
         },
         {
+            "condition": "[parameters('createPublicIP')]",
             "type": "Microsoft.Network/networkInterfaces",
-            "name": "[variables('sourceNicName')]",
+            "name": "[concat(variables('sourceNicName'), 'public')]",
             "apiVersion": "2018-02-01",
             "location": "[parameters('location')]",
-            "properties":
-            {
+            "properties": {
                 "ipConfigurations": [
                     {
                         "name": "ipconfig",
-                        "properties":
-                        {
+                        "properties": {
+                            "publicIPAddress": {
+                                "id": "[resourceId ('Microsoft.Network/publicIPAddresses/', variables('publicIPSourceServer'))]"
+                            },
                             "privateIPAllocationMethod": "Dynamic",
-                            "subnet":
-                            {
+                            "subnet": {
+                                "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('adVNet'), variables('defaultSubnetName'))]"
+                            },
+                            "primary": true,
+                            "privateIPAddressVersion": "IPv4"
+                        }
+                    }
+                ],
+                "enableAcceleratedNetworking": false,
+                "enableIPForwarding": false
+            },
+            "dependsOn": [
```

Comment:
move all of the dependsOn above the properties object in the resources

Reply:
Done